futaba/cogs/pingable/core.py

```diff
@@ -0,0 +1,94 @@
+#
+# cogs/pingable/core.py
+#
+# futaba - A Discord Mod bot for the Programming server
+# Copyright (c) 2017-2020 Jake Richardson, Ammon Smith, jackylam5
+#
+# futaba is available free of charge under the terms of the MIT
+# License. You are free to redistribute and/or modify it under those
+# terms. It is distributed in the hopes that it will be useful, but
+# WITHOUT ANY WARRANTY. See the LICENSE file for more details.
+#
+
+"""
+Cog for pingable helper roles
+"""
+
+import logging
+from datetime import datetime, timedelta
+
+import discord
+from discord.ext import commands
+from futaba.permissions import mod_perm
+from futaba.exceptions import CommandFailed
+from futaba.utils import fancy_timedelta
+from ..abc import AbstractCog
+
+logger = logging.getLogger(__name__)
+
+__all__ = ["Pingable"]
+
+
+class Pingable(AbstractCog):
+    __slots__ = ("journal",)
+
+    def __init__(self, bot):
+        super().__init__(bot)
+        self.journal = bot.get_broadcaster("/pingable")
+        self.cooldowns = {}
```

Comment:
Good idea. I will implement it.

Reply:
> Importing a third party library for an expiring map
> a background coroutine just clean out stale entries

both seem like overkill. just clean them out when you add stuff

Reply:
@raylu Done. I doubt that looping over the dictionary every time someone successfully pings the role will have a big performance impact as well, because I estimate the dictionary will be only around 1000 items or so.